File: src/ServerPulse.Backend/AuthenticationApi.Dtos/AccessTokenDataDto.cs
namespace AuthenticationApi.Dtos
{
    public class AccessTokenDataDto
    {
        public string AccessToken { get; set; } = string.Empty;
        public string RefreshToken { get; set; } = string.Empty;
        public DateTime? RefreshTokenExpiryDate { get; set; }
    }
}

File: src/ServerPulse.Frontend/src/app/modules/authentication/domain/dtos/accessTokenDataDto.ts
import { AccessTokenData } from "../..";

export interface AccessTokenDataDto {
    accessToken: string;
    refreshToken: string;
    refreshTokenExpiryDate: Date;
}

export function mapAuthTokenResponseToAuthToken(response: AccessTokenDataDto): AccessTokenData {
    return {
        accessToken: response?.accessToken,
        refreshToken: response?.refreshToken,
        refreshTokenExpiryDate: new Date(response?.refreshTokenExpiryDate)
    }
}